crypto/fipsmodule/rand/urandom_test.cc

```diff
@@ -471,39 +473,62 @@ static std::vector<Event> TestFunctionPRNGModel(unsigned flags) {
 
   const size_t kSeedLength = CTR_DRBG_ENTROPY_LEN;
   const size_t kAdditionalDataLength = 32;
-
-  if (!have_rdrand()) {
-    if ((!have_fork_detection() && !sysrand(true, kAdditionalDataLength)) ||
-        // Initialise CRNGT. In FIPS mode we use a blocking sysrand call for
-        // a personalization string, while in non-FIPS mode the same call is
-        // used for generating entropy for a seed. Note that the length
-        // of the personalization string is the same as that of the seed.
-        !sysrand(true, kSeedLength) ||
-        // Second entropy draw.
-        (!have_fork_detection() && !sysrand(true, kAdditionalDataLength))) {
+  const size_t kPersonalizationStringLength = CTR_DRBG_ENTROPY_LEN;
+  const bool kHaveRdrand = have_rdrand();
+  const bool kHaveFastRdrand = have_fast_rdrand();
+  const bool kHaveForkDetection = have_fork_detection();
+
+  // Additional data might be drawn on each invocation of RAND_bytes(). In case
+  // it is and there is no rdrand at all, the call is blocking. In the case
+  // where there is at least a "slow" rdrand, first a non-blocking call to
+  // system random is performed followed by an rdrand call.
+
+  // First call to RAND_bytes() will do two things:
+  // * maybe draw for additional data
+  // * seed the CTR-DRBG
+  // First is deciding on additional data.
+  if (!kHaveRdrand || !kHaveFastRdrand) {
+    if (!kHaveForkDetection) {
+      // If no rdrand, we use additional data if fork detection is not enabled.
+      if (!sysrand(!kHaveRdrand, kAdditionalDataLength)) {
+        return ret;
+      }
+    }
+  }
+  // Now the entropy for seeding.
+  if (kIsFIPS) {
+    // In FIPS mode we use Jitter Entropy for the seed but Jitter is not modeled
```

Comment:
I'm not too well-versed in this. What does "Jitter is not modeled" mean here :)?

Reply:
The call to retrieve entropy from Jitter is not part of the model because it can't be picked up by the tracer.

Reply:
Ahh got it, thanks!